test/uk/gov/hmrc/payetaxcalculatorfrontend/model/TaxResultSpec.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.payetaxcalculatorfrontend.model
+
+import uk.gov.hmrc.payeestimator.domain.TaxCalc
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.payetaxcalculatorfrontend.model.TaxResult._
+
+/**
+  * Created by paul on 02/02/17.
+  */
+class TaxResultSpec extends UnitSpec {
+
+  "Extracting Tax Code from user response" should {
+
+    "return tax code that a user has provided" in {
+      extractTaxCode(QuickCalcAggregateInput(Some(UserTaxCode(true,Some("K452"))), None, None)) shouldBe "K452"
+    }
+
+    "return default tax code: 1150L if a user does not provide one" in {
+      extractTaxCode(QuickCalcAggregateInput(Some(UserTaxCode(false,None)), None, None)) shouldBe "1150L"
+    }
+  }
+
+  "Extracting Over65 answer from user response" should {
+
+    "return true if the response is user is over 65" in {
+      extractOver65(QuickCalcAggregateInput(None, Some(Over65(true)), None)) shouldBe "true"
+    }
+
+    "return false if the reponse is user is not over 65" in {
+      extractOver65(QuickCalcAggregateInput(None, Some(Over65(false)), None)) shouldBe "false"
+    }
+
+    //Todo unsure about this?
```

Comment:
We could throw an exception here instead as it should never be None at this point